Add Vault guides (#345)
lukso-network/docs: docs/guides/vault/create-a-vault.md

---
sidebar_label: 'Create a Vault'
sidebar_position: 1
---

# Create a Vault

This guide will teach you how to deploy an **[LSP9Vault](../../standards/smart-contracts/lsp9-vault.md)** contract. This contract can be used to **hold assets** such as tokens and NFTs. Also can be used with a [UniversalProfile](../../standards/universal-profile/introduction.md) and a [KeyManager](../../standards/universal-profile/lsp6-key-manager.md) to **restrict some addresses** (protocols, friends, etc..) to execute and setData on it, instead of setting or executing directly on the profile.

> ![Guide - How to create an LSP9Vault](../../../static/img/guides/LSP9VaultGuide.jpeg)

Reviewer: ```suggestion
![Guide - How to create an LSP9Vault](/img/guides/LSP9VaultGuide.jpeg)
```

## Deploy an LSP9Vault contract

> Check **[previous guides](../universal-profile/transfer-lyx.md#setup)** to learn how to connect to LUKSO's L16 network and load your EOA.

Reviewer: ```suggestion
Check **[previous guides](../universal-profile/create-profile#step-1---create-an-eoa)** to learn how to connect to LUKSO's L16 network and load your EOA.
```

This part explains how to create an address - the Transfert LYX doc doesn't explain it.

```typescript title="Deploying the vault"

> import LSP9Vault from '@lukso/lsp-smart-contracts/artifacts/LSP9Vault.json';
> 
> // create an instance
> let myVault = new web3.eth.Contract(LSP9Vault.abi, {
>   gas: 5_000_000,
>   gasPrice: '1000000000',
> });
> 
> // deploy the vault contract
> await myVault
>   .deploy({
>     data: LSP9Vault.bytecode,
>     arguments: ["0x.."] // address of the UniversalProfile 
>   })
>   .send({
>     from: myEOA.address,
>   });

Reviewer: I suggest to add the line that creates the web3 variable - so users can't use this code with copy/paste.

I just did it and then I'm like "oh no I'm missing the web3, where do I get it?" and then I have to go to other docs.

Let's simply add the web3 here + a line to explain how to install the libs.

```suggestion
import Web3 from 'web3';
import LSP9Vault from '@lukso/lsp-smart-contracts/artifacts/LSP9Vault.json';

const web3 = new Web3('https://rpc.l16.lukso.network');

const PRIVATE_KEY = '0x...'; // your EOA private key
const myEOA = web3.eth.accounts.privateKeyToAccount(PRIVATE_KEY);

// create an instance
const myVault = new web3.eth.Contract(LSP9Vault.abi, {
  gas: 5_000_000,
  gasPrice: '1000000000',
});

// deploy the vault contract
await myVault
  .deploy({
    data: LSP9Vault.bytecode,
    arguments: ["0x.."] // address of the UniversalProfile 
  })
  .send({
    from: myEOA.address,
  });
```

for the new Contract line - i have a TS error:

![image](https://user-images.githubusercontent.com/477945/183431612-b251ed30-2d61-4b56-a2bb-2bf8c18985f6.png)

I tihnk you should move the params from the Contract instantiation to the deploy line instead.
